-1

Tested at: a26912c8e3b2ca32a4ad90552cad3591d3e545d6


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/58f4ff33cf83c52674ad48fd58e8f07904850760
https://github.com/cms-sw/cmssw/commit/57a74eea0c8c028a658844fe8d1a66bbc8f30fd1
https://github.com/cms-sw/cmssw/commit/d2f7bcb1ae39831eb02f29cbf64354044c162893
https://github.com/cms-sw/cmssw/commit/e796093f8257285a3562fe09d07263f7e5b0b908
https://github.com/cms-sw/cmssw/commit/0ceca72dcc157f6b5f5ae3bc959aa7cedbcc2270
https://github.com/cms-sw/cmssw/commit/902b4221ee6a365a9622c61a381f20003b8d2e47
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16343/16367/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16343/16367/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16343/16367/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  ClangBuild**

* **Build**:

I found an error when building: 
 
 <pre>>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-11-15-2300/src/RecoMET/METAlgorithms/src/significanceAlgo.cc 
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-11-15-2300/src/RecoMET/METAlgorithms/src/SignPFSpecificAlgo.cc 
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-11-15-2300/src/RecoMET/METAlgorithms/src/EcalHaloAlgo.cc 
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-11-15-2300/src/RecoMET/METAlgorithms/src/CaloSpecificAlgo.cc 
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-11-15-2300/src/SimCalorimetry/HcalZeroSuppressionProducers/src/HcalRealisticZS.cc: In member function 'virtual void HcalRealisticZS::produce(edm::Event&, const edm::EventSetup&)':
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-11-15-2300/src/SimCalorimetry/HcalZeroSuppressionProducers/src/HcalRealisticZS.cc:131:19: error: 'HBHEUpgradeDigiCollection' was not declared in this scope
   std::unique_ptr<HBHEUpgradeDigiCollection> zs_hbheUpgrade(new HBHEUpgradeDigiCollection);
                   ^
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-11-15-2300/src/SimCalorimetry/HcalZeroSuppressionProducers/src/HcalRealisticZS.cc:131:44: error: template argument 1 is invalid
   std::unique_ptr<HBHEUpgradeDigiCollection> zs_hbheUpgrade(new HBHEUpgradeDigiCollection);
                                            ^
</pre>

* **Clang**:

I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && scram build -k -j 48 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'


 <pre>                         ^~~~~
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-11-15-2300/src/Geometry/Records/src/HcalSimNumberingRecord.cc 
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-11-15-2300/src/Geometry/Records/src/CaloTopologyRecord.cc 
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-11-15-2300/src/Geometry/Records/src/PZdcRcd.cc 
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-11-15-2300/src/Geometry/Records/src/RPCRecoGeometryRcd.cc 
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-11-15-2300/src/SimCalorimetry/HcalZeroSuppressionProducers/src/HcalRealisticZS.cc:131:19: error: unknown type name 'HBHEUpgradeDigiCollection'; did you mean 'HBHEDigiCollection'?
  std::unique_ptr<HBHEUpgradeDigiCollection> zs_hbheUpgrade(new HBHEUpgradeDigiCollection);
                  ^~~~~~~~~~~~~~~~~~~~~~~~~
                  HBHEDigiCollection
/cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_8_1_X_2016-11-15-2300/src/DataFormats/HcalDigi/interface/HcalDigiCollections.h:20:46: note: 'HBHEDigiCollection' declared here
typedef edm::SortedCollection<HBHEDataFrame> HBHEDigiCollection;
</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/58f4ff33cf83c52674ad48fd58e8f07904850760
https://github.com/cms-sw/cmssw/commit/57a74eea0c8c028a658844fe8d1a66bbc8f30fd1
https://github.com/cms-sw/cmssw/commit/d2f7bcb1ae39831eb02f29cbf64354044c162893
https://github.com/cms-sw/cmssw/commit/e796093f8257285a3562fe09d07263f7e5b0b908
https://github.com/cms-sw/cmssw/commit/0ceca72dcc157f6b5f5ae3bc959aa7cedbcc2270
https://github.com/cms-sw/cmssw/commit/902b4221ee6a365a9622c61a381f20003b8d2e47
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16343/16367/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16343/16367/git-merge-result

--- a/Configuration/PyReleaseValidation/python/relval_standard.py
+++ b/Configuration/PyReleaseValidation/python/relval_standard.py
@@ -279,6 +279,9 @@ workflows[34] = ['', ['QQH1352T','DIGI','RECO','HARVEST']]
 
 workflows[7]  = ['', ['Cosmics','DIGICOS','RECOCOS','ALCACOS','HARVESTCOS']]
 workflows[7.1]= ['', ['CosmicsSPLoose','DIGICOS','RECOCOS','ALCACOS','HARVESTCOS']]
+workflows[7.2] = ['', ['Cosmics_UP17','DIGICOS_UP17','RECOCOS_UP17','ALCACOS_UP17','HARVESTCOS_UP17']]
+workflows[7.3] = ['', ['CosmicsSPLoose_UP17','DIGICOS_UP17','RECOCOS_UP17','ALCACOS_UP17','HARVESTCOS_UP17']]
+
 workflows[8]  = ['', ['BeamHalo','DIGICOS','RECOCOS','ALCABH','HARVESTCOS']]
 workflows[11] = ['', ['MinBias','DIGI','RECOMIN','HARVEST','ALCAMIN']]
 workflows[28] = ['', ['QCD_Pt_80_120','DIGI','RECO','HARVEST']]

--- a/Configuration/PyReleaseValidation/python/relval_steps.py
+++ b/Configuration/PyReleaseValidation/python/relval_steps.py
@@ -569,6 +569,8 @@ steps['ZpTT_1500_8TeVINPUT']={'INPUT':InputInfo(dataSet='/RelValZpTT_1500_8TeV_T
 steps['Cosmics']=merge([{'cfg':'UndergroundCosmicMu_cfi.py','--scenario':'cosmics'},Kby(666,100000),step1Defaults])
 steps['CosmicsSPLoose']=merge([{'cfg':'UndergroundCosmicSPLooseMu_cfi.py','--scenario':'cosmics'},Kby(5000,100000),step1Defaults])
 steps['CosmicsSPLoose_UP15']=merge([{'cfg':'UndergroundCosmicSPLooseMu_cfi.py','--conditions':'auto:run2_mc_cosmics','--scenario':'cosmics'},Kby(5000,500000),step1Up2015Defaults])
+steps['Cosmics_UP17']=merge([{'cfg':'UndergroundCosmicMu_cfi.py','--conditions':'81X_upgrade2017cosmics_realistic_peak_v1','--scenario':'cosmics','--era':'Run2_2017'},Kby(666,100000),step1Defaults])
+steps['CosmicsSPLoose_UP17']=merge([{'cfg':'UndergroundCosmicSPLooseMu_cfi.py','--conditions':'81X_upgrade2017cosmics_realistic_peak_v1','--scenario':'cosmics','--era':'Run2_2017'},Kby(5000,500000),step1Up2015Defaults])
 steps['BeamHalo']=merge([{'cfg':'BeamHalo_cfi.py','--scenario':'cosmics'},Kby(9,100),step1Defaults])
 steps['BeamHalo_13']=merge([{'cfg':'BeamHalo_13TeV_cfi.py','--scenario':'cosmics'},Kby(9,100),step1Up2015Defaults])
 
@@ -974,6 +976,7 @@ steps['DIGI']=merge([step2Defaults])
 steps['DIGICOS']=merge([{'--scenario':'cosmics','--eventcontent':'FEVTDEBUG','--datatier':'GEN-SIM-DIGI-RAW'},stCond,step2Defaults])
 steps['DIGIHAL']=merge([{'--scenario':'cosmics','--eventcontent':'FEVTDEBUG','--datatier':'GEN-SIM-DIGI-RAW'},step2Upg2015Defaults])
 steps['DIGICOS_UP15']=merge([{'--conditions':'auto:run2_mc_cosmics','--scenario':'cosmics','--eventcontent':'FEVTDEBUG','--datatier':'GEN-SIM-DIGI-RAW'},step2Upg2015Defaults])
+steps['DIGICOS_UP17']=merge([{'--conditions':'81X_upgrade2017cosmics_realistic_peak_v1','-s':'DIGI:pdigi_valid,DIGI2RAW','--scenario':'cosmics','--eventcontent':'FEVTDEBUG','--datatier':'GEN-SIM-DIGI-RAW', '--era' : 'Run2_2017'},step2Upg2015Defaults])
 
 steps['DIGIPU1']=merge([PU,step2Defaults])
 steps['DIGIPU2']=merge([PU2,step2Defaults])
@@ -1260,6 +1263,8 @@ steps['RECOPRODUP15']=merge([{ '-s':'RAW2DIGI,L1Reco,RECO,EI,PAT,DQM:DQMOfflineP
 steps['RECOCOS']=merge([{'-s':'RAW2DIGI,L1Reco,RECO,ALCA:MuAlGlobalCosmics,DQM','--scenario':'cosmics'},stCond,step3Defaults])
 steps['RECOHAL']=merge([{'-s':'RAW2DIGI,L1Reco,RECO,ALCA:MuAlBeamHalo+MuAlBeamHaloOverlaps,DQM','--scenario':'cosmics'},step3Up2015Hal])
 steps['RECOCOS_UP15']=merge([{'--conditions':'auto:run2_mc_cosmics','-s':'RAW2DIGI,L1Reco,RECO,ALCA:MuAlGlobalCosmics,DQM','--scenario':'cosmics'},step3Up2015Hal])
+steps['RECOCOS_UP17']=merge([{'--conditions':'81X_upgrade2017cosmics_realistic_peak_v1','-s':'RAW2DIGI,RECO,ALCA:MuAlGlobalCosmics,DQM','--scenario':'cosmics','--era':'Run2_2017'},step3Up2015Hal])
+
 steps['RECOMIN']=merge([{'-s':'RAW2DIGI,L1Reco,RECO,EI,ALCA:SiStripCalZeroBias+SiStripCalMinBias,VALIDATION,DQM'},stCond,step3Defaults])
 steps['RECOMINUP15']=merge([{'-s':'RAW2DIGI,L1Reco,RECO,EI,ALCA:SiStripCalZeroBias+SiStripCalMinBias,VALIDATION,DQM'},step3Up2015Defaults])
 steps['RECOAODUP15']=merge([{'--datatier':'AODSIM,MINIAODSIM,DQMIO','--eventcontent':'AODSIM,MINIAODSIM,DQM'},step3Up2015Defaults]) 
@@ -1372,7 +1377,7 @@ steps['ALCACOS']=merge([{'-s':'ALCA:TkAlCosmics0T+MuAlGlobalCosmics+HcalCalHOCos
 steps['ALCABH']=merge([{'-s':'ALCA:TkAlBeamHalo+MuAlBeamHaloOverlaps+MuAlBeamHalo'},stCond,step4Defaults])
 steps['ALCAHAL']=merge([{'-s':'ALCA:TkAlBeamHalo+MuAlBeamHaloOverlaps+MuAlBeamHalo'},step4Up2015Defaults])
 steps['ALCACOS_UP15']=merge([{'--conditions':'auto:run2_mc_cosmics','-s':'ALCA:TkAlBeamHalo+MuAlBeamHaloOverlaps+MuAlBeamHalo'},step4Up2015Defaults])
-
+steps['ALCACOS_UP17']=merge([{'--conditions':'81X_upgrade2017cosmics_realistic_peak_v1','-s':'ALCA:TkAlBeamHalo+MuAlBeamHaloOverlaps"MuAlBeamHalo','--era':'Run2_2017'},step4Up2015Defaults])
 steps['ALCAHARVD']={'-s':'ALCAHARVEST:BeamSpotByRun+BeamSpotByLumi+SiStripQuality',
                     '--conditions':'auto:run1_data',
                     '--scenario':'pp',
@@ -1504,6 +1509,17 @@ steps['HARVESTCOS_UP15']={'-s'          :'HARVESTING:dqmHarvesting',
                           '--filetype':'DQM',
                           '--era' : 'Run2_2016',
                           }
+steps['HARVESTCOS_UP17']={'-s'          :'HARVESTING:dqmHarvesting',
+                          '--conditions':'81X_upgrade2017cosmics_realistic_peak_v1',
+                          '--mc'        :'',
+                          '--filein'    :'file:step3_inDQM.root',
+                          '--scenario'    :'cosmics',
+                          '--filein':'file:step3_inDQM.root', # unnnecessary
+                          '--filetype':'DQM',
+                          '--era' : 'Run2_2017'
+                          }
+
+		  
 steps['HARVESTFS']={'-s':'HARVESTING:validationHarvesting',
                    '--conditions':'auto:run1_mc',
                    '--mc':'',
@@ -1776,6 +1792,7 @@ for year,k in [(year,k) for year in upgradeKeys for k in upgradeKeys[year]]:
                                       }
     if cust!=None : upgradeStepDict['RecoFull'][k]['--customise']=cust
     if era is not None: upgradeStepDict['RecoFull'][k]['--era']=era
+
 
     if k2 in PUDataSets:
         upgradeStepDict['RecoFullPU'][k]=merge([PUDataSets[k2],upgradeStepDict['RecoFull'][k]])

--- a/Configuration/PyReleaseValidation/python/relval_steps.py
+++ b/Configuration/PyReleaseValidation/python/relval_steps.py
@@ -976,7 +976,7 @@ steps['DIGI']=merge([step2Defaults])
 steps['DIGICOS']=merge([{'--scenario':'cosmics','--eventcontent':'FEVTDEBUG','--datatier':'GEN-SIM-DIGI-RAW'},stCond,step2Defaults])
 steps['DIGIHAL']=merge([{'--scenario':'cosmics','--eventcontent':'FEVTDEBUG','--datatier':'GEN-SIM-DIGI-RAW'},step2Upg2015Defaults])
 steps['DIGICOS_UP15']=merge([{'--conditions':'auto:run2_mc_cosmics','--scenario':'cosmics','--eventcontent':'FEVTDEBUG','--datatier':'GEN-SIM-DIGI-RAW'},step2Upg2015Defaults])
-steps['DIGICOS_UP17']=merge([{'--conditions':'81X_upgrade2017cosmics_realistic_peak_v1','-s':'DIGI:pdigi_valid,DIGI2RAW','--scenario':'cosmics','--eventcontent':'FEVTDEBUG','--datatier':'GEN-SIM-DIGI-RAW', '--era' : 'Run2_2017'},step2Upg2015Defaults])
+steps['DIGICOS_UP17']=merge([{'--conditions':'81X_upgrade2017cosmics_realistic_peak_v1','-s':'DIGI:pdigi_valid,L1,DIGI2RAW,HLT:@fake','--scenario':'cosmics','--eventcontent':'FEVTDEBUG','--datatier':'GEN-SIM-DIGI-RAW', '--era' : 'Run2_2017'},step2Upg2015Defaults])
 
 steps['DIGIPU1']=merge([PU,step2Defaults])
 steps['DIGIPU2']=merge([PU2,step2Defaults])
@@ -1263,7 +1263,7 @@ steps['RECOPRODUP15']=merge([{ '-s':'RAW2DIGI,L1Reco,RECO,EI,PAT,DQM:DQMOfflineP
 steps['RECOCOS']=merge([{'-s':'RAW2DIGI,L1Reco,RECO,ALCA:MuAlGlobalCosmics,DQM','--scenario':'cosmics'},stCond,step3Defaults])
 steps['RECOHAL']=merge([{'-s':'RAW2DIGI,L1Reco,RECO,ALCA:MuAlBeamHalo+MuAlBeamHaloOverlaps,DQM','--scenario':'cosmics'},step3Up2015Hal])
 steps['RECOCOS_UP15']=merge([{'--conditions':'auto:run2_mc_cosmics','-s':'RAW2DIGI,L1Reco,RECO,ALCA:MuAlGlobalCosmics,DQM','--scenario':'cosmics'},step3Up2015Hal])
-steps['RECOCOS_UP17']=merge([{'--conditions':'81X_upgrade2017cosmics_realistic_peak_v1','-s':'RAW2DIGI,RECO,ALCA:MuAlGlobalCosmics,DQM','--scenario':'cosmics','--era':'Run2_2017'},step3Up2015Hal])
+steps['RECOCOS_UP17']=merge([{'--conditions':'81X_upgrade2017cosmics_realistic_peak_v1','-s':'RAW2DIGI,L1Reco,RECO,ALCA:MuAlGlobalCosmics,DQM','--scenario':'cosmics','--era':'Run2_2017'},step3Up2015Hal])
 
 steps['RECOMIN']=merge([{'-s':'RAW2DIGI,L1Reco,RECO,EI,ALCA:SiStripCalZeroBias+SiStripCalMinBias,VALIDATION,DQM'},stCond,step3Defaults])
 steps['RECOMINUP15']=merge([{'-s':'RAW2DIGI,L1Reco,RECO,EI,ALCA:SiStripCalZeroBias+SiStripCalMinBias,VALIDATION,DQM'},step3Up2015Defaults])

--- a/RecoLocalCalo/Configuration/python/RecoLocalCalo_Cosmics_cff.py
+++ b/RecoLocalCalo/Configuration/python/RecoLocalCalo_Cosmics_cff.py
@@ -7,48 +7,88 @@ from RecoLocalCalo.Configuration.ecalLocalRecoSequenceCosmics_cff import *
 from RecoLocalCalo.EcalRecAlgos.EcalSeverityLevelESProducer_cfi import *
 
 #defines a sequence ecalLocalRecoSequence
+
 #
 # Hcal part
 #
-# calo geometry
-#
-# changed by tommaso. now the calibrations are read from Configuration/StaqndardSequences/data/*Conditions.cff
-#
-# HCAL calibrations
-#include "CalibCalorimetry/HcalPlugins/data/hardwired_conditions.cfi"
 #HCAL reconstruction
-from RecoLocalCalo.Configuration.hcalLocalReco_cff import *
+import RecoLocalCalo.Configuration.hcalLocalReco_cff as _hcalLocalReco_cff
 #
-# sequence CaloLocalReco and CaloGlobalReco
+# sequence CaloLocalReco
 #
+hbheprereco = _hcalLocalReco_cff._default_hbheprereco.clone(
+    puCorrMethod = 0,
+    firstSample = 0,
+    samplesToAdd = 10,
+    correctForTimeslew = False,
+    correctForPhaseContainment = False,
+    tsFromDB = False,
+    recoParamsFromDB = cms.bool(False),
+)
+hfreco = _hcalLocalReco_cff._default_hfreco.clone(
+    firstSample = 0,
+    samplesToAdd = 10, ### min(10,size) in the algo
+    correctForTimeslew = False,
+    correctForPhaseContainment = False,
+    tsFromDB = False,
+    recoParamsFromDB = cms.bool(False),
+    digiTimeFromDB = False,
+)
+horeco = _hcalLocalReco_cff.horeco.clone(
+    firstSample = 0,
+    samplesToAdd = 10,
+    correctForTimeslew = False,
+    correctForPhaseContainment = False,
+    tsFromDB = False,
+    recoParamsFromDB = cms.bool(False),
+)
+zdcreco = _hcalLocalReco_cff.zdcreco.clone(
+#    firstSample = 1,
+#    samplesToAdd = 8,
+    correctForTimeslew = True,
+    correctForPhaseContainment = True,
+    correctionPhaseNS = 10.,
+)
+
+# 2017 customs
+from Configuration.Eras.Modifier_run2_HE_2017_cff import run2_HE_2017
+from Configuration.Eras.Modifier_run2_HF_2017_cff import run2_HF_2017
+
+_phase1_hbheprereco = _hcalLocalReco_cff._phase1_hbheprereco.clone(
+    tsFromDB = cms.bool(False),
+    recoParamsFromDB = cms.bool(False),
+    algorithm = dict(
+        useM2 = cms.bool(False),
+        useM3 = cms.bool(False),
+        firstSampleShift = cms.int32(-1000),
+        samplesToAdd = cms.int32(10),
+        correctForPhaseContainment = cms.bool(False),
+    )
+)
+
+_phase1_hfreco = _hcalLocalReco_cff._phase1_hfreco.clone(
+    algorithm = dict(
+        Class = cms.string("HFSimpleTimeCheck"),
+        rejectAllFailures = cms.bool(False),
+    )
+)
+
+
+run2_HE_2017.toReplaceWith(hbheprereco, _phase1_hbheprereco )
+run2_HF_2017.toReplaceWith(hfreco, _phase1_hfreco )
+
+hfprereco = _hcalLocalReco_cff.hfprereco.clone(
+    sumAllTimeSlices = cms.bool(True)
+)
+
+# redefine hcal sequence
+hcalLocalRecoSequence = cms.Sequence(hbheprereco+hfreco+horeco+zdcreco)
+
+_phase1_hcalLocalRecoSequence = hcalLocalRecoSequence.copy()
+_phase1_hcalLocalRecoSequence.insert(0,hfprereco)
+run2_HF_2017.toReplaceWith(hcalLocalRecoSequence, _phase1_hcalLocalRecoSequence)
+
 calolocalrecoCosmics = cms.Sequence(ecalLocalRecoSequenceCosmics+hcalLocalRecoSequence)
-hbheprereco.puCorrMethod = 0 
-hbheprereco.firstSample = 0
-hbheprereco.samplesToAdd = 10
-hbheprereco.correctForTimeslew = False
-hbheprereco.correctForPhaseContainment = False
-horeco.firstSample = 0
-horeco.samplesToAdd = 10
-horeco.correctForTimeslew = False
-horeco.correctForPhaseContainment = False
-hfreco.firstSample = 0
-hfreco.samplesToAdd = 10 ### min(10,size) in the algo
-hfreco.correctForTimeslew = False
-hfreco.correctForPhaseContainment = False
-#--- special temporary DB-usage unsetting 
-hbheprereco.tsFromDB = False
-hfreco.tsFromDB = False
-horeco.tsFromDB = False
-hfreco.digiTimeFromDB = False
-hbheprereco.recoParamsFromDB = cms.bool(False)
-horeco.recoParamsFromDB      = cms.bool(False)
-hfreco.recoParamsFromDB      = cms.bool(False)
-#zdcreco.firstSample = 1
-#zdcreco.samplesToAdd = 8
-zdcreco.correctForTimeslew = True
-zdcreco.correctForPhaseContainment = True
-zdcreco.correctionPhaseNS = 10.
-#caloglobalreco = cms.Sequence(hcalGlobalRecoSequence)
 
 #
 # R.Ofierzynski (29.Oct.2009): add NZS sequence

--- a/RecoLocalCalo/Configuration/python/hcalLocalReco_cff.py
+++ b/RecoLocalCalo/Configuration/python/hcalLocalReco_cff.py
@@ -13,6 +13,10 @@ from RecoLocalCalo.HcalRecProducers.hfprereco_cfi import hfprereco
 from RecoLocalCalo.HcalRecProducers.HFPhase1Reconstructor_cfi import hfreco as _phase1_hfreco
 from RecoLocalCalo.HcalRecProducers.HBHEPhase1Reconstructor_cfi import hbheprereco as _phase1_hbheprereco
 
+# copies for cosmics
+_default_hbheprereco = hbheprereco.clone()
+_default_hfreco = hfreco.clone()
+
 _phase1_hcalLocalRecoSequence = hcalLocalRecoSequence.copy()
 _phase1_hcalLocalRecoSequence.insert(0,hfprereco)
 

